### ISSUE TITLE
`Loader` type

### ISSUE THREAD

**Issue by rhys-vdw** [rhys-vdw/atlas#15]
Enable the following interfaces with a `Loader` type.

```js
// Where `relation` is either a `Relation` instance, or the string identifier of a
// `Relation` instance registered on the mapper (via `Mapper#relations()`).
Mapper.load(related(relation)).into(records);
Mapper.with(related(relation)).fetch(); // Calls `load(...).into(...)` internally.
```

`Loader` holds record state between calls, which is something to be avoided for `Mapper` etc.

The `Loader` instance is created internally and looks something like this:
```js
import { first, flatten, isArray } from 'lodash';

class Loader {
  constructor(related) {
    this.related = related;
  }

  into(...records) {
    // Are we assigning to a single record, or to an array of records?
    const isParentSingle = records.length === 1 && !isArray(first(records));

    // Simplify the contract for the `Related` by always providing an array.
    const parents = flatten(records);

    // Now refine the query to target only records related to the parent and fetch them.
    return related.targeting(parents).fetch().then(relatedRecords => {

      // The `Related` class also handles assignment.
      const assignedParents = related.assignRelated(parents, ensureArray(relatedRecords));

      // Return either the single instance or array (to match input).
      return isParentSingle ? first(assignedParents) : assignedParents;
    });
  }
}
```

Other considerations:
How does one handle something like this:
```js
Mapper.load(related('friends', 'family', 'residence')).into(...)
```

Currenty `Related#assignRelated` should treat the `parents` array and objects as immutable, so it would be a shame to have to copy the entire array for each relation to be loaded in.

**Comment by rhys-vdw** [comment]
Closed by #17

**Comment by rhys-vdw** [comment]
(This class is called `EagerLoader` now)
